proposals/0540-jsem.rst

```diff
@@ -246,6 +258,99 @@ with the semaphore: instead of systematically releasing a token to the semaphore
 once a job is done, we can instead re-use this token if we have other pending
 jobs. This prioritises the compilation of a single unit.
 
+Abstract Semaphore Interface
```

Comment:
This subsection is at least partly about (a) history and (b) implementation details.   None of it is part of the proposed specification.

Maybe move 
* back-compt stuff to a section on back-compat (see [template](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0000-template.rst)) .  Including the current 2.6 
* implementation stuff  (incl the current 2.5) to a Section on Implementation.  (Proposals often have such a section.)